app/controllers/invites_controller.rb

```diff
@@ -16,22 +16,24 @@ def show
     expires_now
 
     invite = Invite.find_by(invite_key: params[:id])
-    if invite.present? && !invite.expired? && !invite.redeemed?
+    if invite.present? && invite.redeemable?
       store_preloaded("invite_info", MultiJson.dump(
         invited_by: UserNameSerializer.new(invite.invited_by, scope: guardian, root: false),
         email: invite.email,
         username: UserNameSuggester.suggest(invite.email),
         is_invite_link: invite.is_invite_link?
       ))
 
+      secure_session["invite-key"] = invite.invite_key
```

Comment:
I'm not familiar with `secure_session` but does it expire? Do we need to clean this up?

Reply:
It is cleaned up in `sessions_controller`. From the comments RE: expiry:

```
# session that is not stored in cookie, expires after 1.hour unconditionally
```

Reply:
Ahh OK because this means that visiting an invites redemption page will tie the user to the invite for an hour.

Reply:
Yes if they don't redeem it or visit the invite page with a different link